addons/udes_stock_refactoring/models/stock_picking_batch.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+from odoo import fields, models, api
+
+
+class StockPickingBatch(models.Model):
+    _inherit = "stock.picking.batch"
+
+    @api.depends("picking_ids", "picking_ids.picking_type_id")
+    def _compute_picking_type(self):
+        """Compute the picking types from the linked pickings"""
+        for batch in self:
+            if batch.picking_ids:
+                batch.picking_type_ids = batch.picking_ids.picking_type_id
+            elif not isinstance(batch.id, models.NewId):
+                # If the picking ids are empty use the stored picking type ids
+                batch.picking_type_ids = batch.read(["picking_type_ids"])[0]["picking_type_ids"]
```

Comment:
this last one is it not empty? I'm clearly missing something

Reply:
Really not sure why this was done, have not been able to trigger this either. Tempted to remove this elif.

Reply:
It was needed in O11 for other reasons, and probably not needed in O13. You can remove it. But it is mandatory to set a value on O13, so you will sill need an else.